### PR TITLE
Provide users with more customization when creating a new bot.

### PR DESCRIPTION
**Pull request by OGKevin**
User are now able to create a bot with debug enabled by default.

The debug part is use full  when `NewBotAPI()` is failing with the message 

```
http.2.0bttrvjk5f4e@linuxkit-025000000001    | 2018/11/12 12:15:38 created msg
http.2.0bttrvjk5f4e@linuxkit-025000000001    | 2018/11/12 12:15:38 getting bot
http.2.0bttrvjk5f4e@linuxkit-025000000001    | 2018/11/12 12:15:38 error is about to happen
http.2.0bttrvjk5f4e@linuxkit-025000000001    | 2018/11/12 12:15:38 Not Found (this one 🤷🏾‍♂️)
```

By being able to enale debug when creating the bot user are able to debug the following code:

```
		b, err := tgbotapi.NewBotAPI(os.Getenv("TELEGRAM_BOT_TOKEN"))

		if err != nil {
			log.Print("error is about to happen")
			log.Fatal(err)
		}
```

### CODE REVIEW

**Comment by OGKevin** [comment]
@Syfaro @zhulik any feedback?

**Comment by zhulik** [comment]
LGTM, thank you!

File: bot.go [hunk 1]
```diff
@@ -34,19 +34,36 @@ type BotAPI struct {
...
 // It requires a token, provided by @BotFather on Telegram.
 func NewBotAPIWithClient(token string, client *http.Client) (*BotAPI, error) {
+	return CreateNewBotApi(token, client, false)
+}
+
+// NewBotAPIWithClient creates a new BotAPI instance
+// and enables debug by default.
+//
+// It requires a token, provided by @BotFather on Telegram.
+func NewBotApiWithDebug(token string) (*BotAPI, error) {
```
Comment (Syfaro) [inline bot.go]: This should be `NewBotAPIWithDebug` for consistency. 

```suggestion
func NewBotAPIWithDebug(token string) (*BotAPI, error) {
```

File: bot.go [hunk 2]
```diff
@@ -34,19 +34,36 @@ type BotAPI struct {
...
+	return CreateNewBotApi(token, &http.Client{}, true)
+}
+
+// NewBotAPIWithClient creates a new BotAPI instance
+// and allows you to pass all additional customization that you might need.
+//
+// It requires a token, provided by @BotFather on Telegram.
+func CreateNewBotApi(token string, client *http.Client, debug bool) (*BotAPI, error) {
```
Comment (Syfaro) [inline bot.go]: Same here for the method name.

```suggestion
func CreateNewBotAPI(token string, client *http.Client, debug bool) (*BotAPI, error) {
```

File: bot.go [hunk 3]
```diff
@@ -34,19 +34,36 @@ type BotAPI struct {
...
 
 // NewBotAPIWithClient creates a new BotAPI instance
 // and allows you to pass a http.Client.
 //
 // It requires a token, provided by @BotFather on Telegram.
 func NewBotAPIWithClient(token string, client *http.Client) (*BotAPI, error) {
+	return CreateNewBotApi(token, client, false)
+}
+
+// NewBotAPIWithClient creates a new BotAPI instance
```
Comment (Syfaro) [inline bot.go]: Comment needs to be fixed.

```suggestion
// NewBotAPIWithDebug creates a new BotAPI instance
```

File: bot.go [hunk 4]
```diff
@@ -34,19 +34,36 @@ type BotAPI struct {
...
+// and enables debug by default.
+//
+// It requires a token, provided by @BotFather on Telegram.
+func NewBotApiWithDebug(token string) (*BotAPI, error) {
+	return CreateNewBotApi(token, &http.Client{}, true)
+}
+
+// NewBotAPIWithClient creates a new BotAPI instance
```
Comment (Syfaro) [inline bot.go]: Again, comment. 

```suggestion
// CreateNewBotAPI creates a new BotAPI instance
```

**Comment by AndresCidoncha** [comment]
This feature looks really useful to me. Are you working on it @OGKevin? If not, I can change the code with the @Syfaro suggestions.

**Comment by OGKevin** [comment]
@AndresCidoncha ah, I think i've missed the notification about the change request 😮 ! I'll pick this up today!

**Comment by OGKevin** [comment]
@Syfaro addressed your comments.

**Comment by OGKevin** [comment]
@Syfaro any updates?

**Comment by OGKevin** [comment]
@Syfaro @zhulik Can we get this merged?

**Comment by Syfaro** [comment]
I'm sorry to have had you go through all the work of fixing this up, but I think the approach as seen in #206 of using a `Config` struct makes more sense as it allows further expansion without having to add so many new parameters.

**Pull request closed without merging**